service_discovery_manager/Component/mmNM/tunneling/TunDrv.cpp

```diff
@@ -55,134 +55,134 @@
-    DPRINT(COMM, DEBUG_ERROR, "Cannot Open Tunneling Driver [/dev/net/tun]\n");
-    return -1;
-  }
-#endif
-
-  memset(&ifr, 0, sizeof(ifr));
-  ifr.ifr_flags = IFF_TUN | IFF_NO_PI;
-
-  if (*dev) {
-    DPRINT(COMM, DEBUG_ERROR, "cp ifname [%s]\n", dev);
-    strncpy(ifr.ifr_name, dev, IFNAMSIZ);
-  }
-
-  if (ioctl(fd, TUNSETIFF, (void*)&ifr) < 0) {
-    if (errno == EBADFD) {
-      /* Try old ioctl */
-      if (ioctl(fd, OTUNSETIFF, (void*)&ifr) < 0) {
-        DPRINT(COMM, DEBUG_ERROR, "Cannot Set Driver OTUNSETIFF\n");
-        return -1;
-      }
-    } else {
-      DPRINT(COMM, DEBUG_ERROR, "Cannot Set Driver OTUNSETIFF\n");
-      return -1;
-    }
-  }
-  strcpy(dev, ifr.ifr_name);
-  DPRINT(COMM, DEBUG_ERROR, "Set IFF [%s]\n", dev);
-
-  char cmd_buf[1024];
-
-  memset(cmd_buf, 0, 1024);
-  sprintf(cmd_buf, "ifconfig %s inet %s", ifr.ifr_name, pb_addr);
-  DPRINT(COMM, DEBUG_INFO, "sh command : %s\n", cmd_buf);
-  system(cmd_buf);
-
-  memset(cmd_buf, 0, 1024);
-  sprintf(cmd_buf, "route add -net 10.10.10.0 netmask 255.255.255.0 gw %s",
-          pb_addr);
-  DPRINT(COMM, DEBUG_INFO, "sh command : %s\n", cmd_buf);
-  system(cmd_buf);
-
-  /*
-          for(int i=1;i<=255; i++)
-          {
-                  memset(cmd_buf,0,1024);
-                  sprintf(cmd_buf,"route add 10.10.10.%d dev %s", i,
-     ifr.ifr_name);
-                  DPRINT(COMM,DEBUG_INFO,"sh command : %s\n",cmd_buf);
-                  system(cmd_buf);
-          }
-  */
-
-  // route add 10.10.10.3 tun0
-  return fd;
-#else
-  return 0;
-#endif
-}
-
-int CTunDrv::Close(int fd) {
-#ifndef WIN32
-  return close(fd);
-#else
-  return 0;
-#endif
-}
-
-int CTunDrv::Read(int fd, char* buf, int len) {
-#ifndef WIN32
-  int nReadByte = 0;
-  int count = 10;
-
-  while (--count) {
-    nReadByte = read(fd, buf, len);
-
-    if (nReadByte > 0)
-      return nReadByte;
-
-    usleep(1000);
-    printf("TunDrv Read Sleep(%d)\n", count);
-  }
-
-  return -1;
-#else
-  return -1;
-#endif
-}
-
-int CTunDrv::Write(int fd, char* buf, int len) {
-#ifndef WIN32
-  int nWriteByte = 0, total = 0;
- 
-  int count = 10;
-
-  while (--count) {
-    nWriteByte = write(fd, buf, len);
-
-    total += nWriteByte;
-
-    if (total >= len)
-      return total;
-
-    usleep(1000);
-    printf("TunDrv Write Sleep(%d)\n", count);
-  }
-
-  return -1;
-#else
-  return -1;
-#endif
-}
-
+#define TAP_IOCTL_CONFIG_TUN            _TAP_IOCTL(10) 
+#endif
+
+CTunDrv::CTunDrv() {}
+
+CTunDrv::~CTunDrv() {}
+
+int CTunDrv::Open(char* dev, char* pb_addr) {
+
+#ifndef WIN32
+  struct ifreq ifr;
+  int fd;
+#ifdef ANDROID
+  if ((fd = open("/dev/tun", O_RDWR | O_NONBLOCK)) < 0) {
+    DPRINT(COMM, DEBUG_ERROR, "Cannot Open Tunneling Driver [/dev/tun]\n");
+    return -1;
+  }
+#else
+  if ((fd = open("/dev/net/tun", O_RDWR | O_NONBLOCK)) < 0) {
+    DPRINT(COMM, DEBUG_ERROR, "Cannot Open Tunneling Driver [/dev/net/tun]\n");
+    return -1;
+  }
+#endif
+
+  memset(&ifr, 0, sizeof(ifr));

```

Comment:
I don't see a definition for 'ifr' if WIN32 is defined.

Reply:
Currently, the source code of under tunneling is not included in win32 project (Plz Look at the vs project file).
The reason why this change included  at this commit is just changing of file mode. (previously abnormally file has +x attribute. so removed "x" attribute. )
Windows implementation for tunneling is not prepared yet. for supporting windows tunneling requires totally new implementation from virtual network adapter to virtual network adapter based device I/O.
Till that implementation will be ready, plz, skip Windows tunneling supporting with this code base.